### PR TITLE
Added Dockerfile, .dockerignore

### LINKED ISSUE #12: Create Dockerfile
Dockerfile to be used in pipeline and deploy

Acceptance Criteria
- [x] PR merged to master

### PR DESCRIPTION
**Pull request by Ts1n**
Closes #12

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+COPY . .
+
+# Gradle build with daemon disabled
+RUN ./gradlew build --no-daemon --stacktrace
+
+FROM openjdk:${RUN_TAG} AS run-env
+WORKDIR /app
+
+# Copy generated .jar
+COPY --from=build-env /src/build/libs/*.jar i-on-integration.jar
```
Comment (pmhsfelix) [inline Dockerfile]: See https://github.com/i-on-project/example-ci-cd/blob/experiment0/Dockerfile for a layered way of building the docker file, avoiding creating a single layer containing both the dependent libs (which can be large, but don't change often) and app classes (which are small and change often).

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ARG BUILD_TAG=8-jdk-alpine
+ARG RUN_TAG=8-jre-alpine
+
+FROM openjdk:${BUILD_TAG} AS build-env
+WORKDIR /src
+
+# Copy src .dockerignore prevents copy of unnecessary files
+COPY . .
```
Comment (pmhsfelix) [inline Dockerfile:8]: Does this only copy the `src` folder? We also need files in the root folder and other nested folders (e.g. checkstyle) to perform the build.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+# Gradle build with daemon disabled
+RUN ./gradlew build --no-daemon --stacktrace
```
Comment (pmhsfelix) [inline Dockerfile]: What do we gain by doing the build inside docker?

**Review by pmhsfelix — COMMENTED** [review]
Waiting for answers to questions, before approving.

File: Dockerfile (hunk 2 above)
Comment (Ts1n) [reply Dockerfile:8]: This copies the whole `root folder` of the project (`src folder`, `gradle` and others) ignoring the files mention on `.dockerignore` to the current folder (`WORKDIR`) of the container

File: Dockerfile (hunk 3 above)
Comment (Ts1n) [reply Dockerfile]: We remove the need for the build machine to have the necessary build tools. I believe it to be more portable.

File: Dockerfile (hunk 1 above)
Comment (Ts1n) [reply Dockerfile]: That's a better solution. I've change the Dockerfile.
In the example shown I don't understand the need of creating a new user...

Comment (pmhsfelix) [reply Dockerfile]: Increased security. https://spring.io/guides/gs/spring-boot-docker/ and https://security.stackexchange.com/questions/106860/can-a-root-user-inside-a-docker-lxc-break-the-security-of-the-whole-system

**Review by pmhsfelix — DISMISSED** [review]

**Review by sscosta — APPROVED** [review]

**Pull request merged**